edx_event_bus_kafka/publishing/event_producer.py

```diff
@@ -176,10 +155,11 @@ def inner_to_dict(event_data, ctx=None):  # pylint: disable=unused-argument
         to_dict=inner_to_dict,
     )
 
-    producer_settings.update({
-        'key.serializer': key_serializer,
-        'value.serializer': value_serializer,
-    })
+    if producer_settings is not None:
```

Comment:
If `producer_settings` is `None`, that means the settings weren't valid -- so we'll want to do this check and `return None` as soon as we can.